### PR TITLE
Added new test cases for api's

### PR DESCRIPTION
**Pull request by arpit2438735**
@nomego Have a look to test cases.Meanwhile I am adding more test cases

### CODE REVIEW

**Comment by arpit2438735** [comment]
Looking at broken build

**Comment by plequang** [comment]
I think tests should try to validate the purpose of the element and it's public API.
For example, regarding `_computeBarHeight` function. 
We don't really need to test this function individually, but we rather need to test that when the public property `matchParent` is set to true, then the bottom-bar height should be the same as the height of its parent element.

This could be tested that way:
```html

<test-fixture id="heightManagement">
  <template>
    <div id="heightManagementParent" style="min-height: 100px; max-height: 100px">
      <cosmoz-bottom-bar>
        <fixed-size-item width="50px" height="32px"></fixed-size-item>
      </cosmoz-bottom-bar>
    </div>
  </template>
</test-fixture>
```
```js
suite('height management', function () {
  var heightManagementParent,
      bottomBar;
  
  setup(function () {
    heightManagementParent = fixture('heightManagement');
    bottomBar = Polymer.dom(heightManagementParent).querySelector('cosmoz-bottom-bar');
  });

  test('setting matchParent to true should set bottom-bar height to the height of the parent', function () {
    bottomBar.matchParent = true;
    assert.equal(bottomBar.offsetHeight , 100);
    ...

  test('setting matchParent to true should set matchElementHeight property', function () {
  ...
  // should also test bar height when matchParent is false 

```

That way, if ever you do some refactorying of the internal code of the element, these tests will still be valid.

Same apply to scroll management, which is controlled by the `scroller` property.
When this property is set, then the bar should be visible/active only when scroll position reach bottom of the scroller element.

**Comment by arpit2438735** [comment]
Total make sense

**Comment by arpit2438735** [comment]
@plequang Have a look to new commit.

File: test/cosmoz-bottom-bar.html [hunk 1]
```diff
@@ -205,6 +214,25 @@
...
+			});
+
+			test('setting matchParent to true should set bottom-bar height to the height of the parent', function () {
+				bottomBar.matchParent = true;
+				assert.equal(bottomBar.offsetHeight, 100);
+			});
+
+			test('setting matchParent to false should set bottom-bar less than to the height of the parent', function () {
+				bottomBar.matchParent = false;
+				assert.isBelow(bottomBar.offsetHeight, 100);
```
Comment (arpit2438735) [inline test/cosmoz-bottom-bar.html]: But thats not happening according to logic.That's the bug or problem in documentation?

Comment (plequang) [inline test/cosmoz-bottom-bar.html]: The logic is :

> if (matchParent is true)  then bottom bar height should be the same as the parent element
> else the bottom bar height should be `barHeight`

so the test should use :
```
assert.equal(bottomBar.offsetHeight, bottomBar.barHeight);
```
If that assertion is not OK, then there is a bug.

File: test/cosmoz-bottom-bar.html [hunk 2]
```diff
@@ -230,7 +230,7 @@
 
 			test('setting matchParent to false should set bottom-bar less than to the height of the parent', function () {
 				bottomBar.matchParent = false;
-				assert.isBelow(bottomBar.offsetHeight, 100);
+				assert.equal(bottomBar.offsetHeight, bottomBar.barHeight);
```
Comment (arpit2438735) [inline test/cosmoz-bottom-bar.html]: @plequang Yup it's working

Comment (plequang) [inline test/cosmoz-bottom-bar.html]: Good. 
The test description should be changed.

**Pull request merged**